src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -52,104 +67,834 @@ static ColliderJntSphInit D_809BF3F8 = {
+    this->actor.world.pos.y -= 10.0f;
+    this->actor.shape.rot.y += 0x2000;
+    if (this->actor.world.pos.y < (this->actor.home.pos.y + -200.0f)) {
+        func_809BD7F0(this, globalCtx);
+    }
+    func_809BCF68(this, globalCtx);
+}
+
+void func_809BE518(EnBigokuta* this, GlobalContext* globalCtx) {
+    if (Math_StepToF(&this->actor.world.pos.y, this->actor.home.pos.y, 10.0f)) {
+        this->actor.flags |= 1;
+        func_809BD3F8(this);
+    }
+}
+
+void func_809BE568(EnBigokuta* this) {
+    s32 i;
+    f32 sin = Math_SinS(this->actor.shape.rot.y);
+    f32 cos = Math_CosS(this->actor.shape.rot.y);
+
+    this->collider.elements->dim.worldSphere.center.x =
+        (this->collider.elements->dim.modelSphere.center.z * sin) +
+        (this->actor.world.pos.x + (this->collider.elements->dim.modelSphere.center.x * cos));
+    this->collider.elements->dim.worldSphere.center.z =
+        (this->actor.world.pos.z + (this->collider.elements->dim.modelSphere.center.z * cos)) -
+        (this->collider.elements->dim.modelSphere.center.x * sin);
+    this->collider.elements->dim.worldSphere.center.y =
+        this->collider.elements->dim.modelSphere.center.y + this->actor.world.pos.y;
+
+    for (i = 0; i < ARRAY_COUNT(this->cylinder); i++) {
+        this->cylinder[i].dim.pos.x =
+            this->actor.world.pos.x + sCylinderInit[i].dim.pos.z * sin + sCylinderInit[i].dim.pos.x * cos;
+        this->cylinder[i].dim.pos.z =
+            this->actor.world.pos.z + sCylinderInit[i].dim.pos.z * cos - sCylinderInit[i].dim.pos.x * sin;
+        this->cylinder[i].dim.pos.y = this->actor.world.pos.y;
+    }
+}
+
+void func_809BE798(EnBigokuta* this, GlobalContext* globalCtx) {
+    s16 effectRot;
+    s16 yawDiff;
+
+    if ((this->cylinder[0].base.atFlags & AT_HIT) || (this->cylinder[1].base.atFlags & AT_HIT) ||
+        (this->collider.base.atFlags & AT_HIT)) {
+        this->cylinder[0].base.atFlags &= ~AT_HIT;
+        this->cylinder[1].base.atFlags &= ~AT_HIT;
+        this->collider.base.atFlags &= ~AT_HIT;
+        yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+        if (yawDiff > 0x4000) {
+            effectRot = 0x4000;
+        } else if (yawDiff > 0) {
+            effectRot = 0x6000;
+        } else if (yawDiff < -0x4000) {
+            effectRot = -0x4000;
+        } else {
+            effectRot = -0x6000;
+        }
+        func_8002F71C(globalCtx, &this->actor, 10.0f, this->actor.world.rot.y + effectRot, 5.0f);
+        if (this->actionFunc == func_809BDC08) {
+            func_809BD4A4(this);
+            this->unk_196 = 40;
+        } else if (this->actionFunc == func_809BE3E4) {
+            if ((effectRot * this->unk_194) > 0) {
+                this->unk_194 = 0 - this->unk_194;
+                this->unk_196 += 4;
+            }
+        }
+    }
+}
+// EnBigokuta_CollisionCheck
+void func_809BE8DC(EnBigokuta* this, GlobalContext* globalCtx) {
```

Comment:
```suggestion

void EnBigokuta_CollisionCheck(EnBigokuta* this, GlobalContext* globalCtx) {
```
Might as well name it since its there.